tests/test_eigendistortion.py

```diff
@@ -104,69 +103,44 @@ def test_method_power(self, color):
-
-    def test_lanczos_accuracy(self):
-        n = 30
-        e_vals = (torch.randn(n**2)**2).sort(descending=True)[0]
-        eigen_test_matrix = torch.diag(e_vals)
+    def test_orthog_iter(self):
+        n, k = 30, 10
+        n_chans = 1  # TODO color
         ed = get_synthesis_object(im_dim=n)
-        with pytest.warns(UserWarning) as lanczos_experimental_warning:
-            ed.synthesize(method='lanczos', n_steps=eigen_test_matrix.shape[-1], debug_A=eigen_test_matrix)
+        ed.synthesize(k=k, method='power', max_steps=10)
 
-        assert (e_vals[0]-ed.synthesized_eigenvalues[0]) < 1e-2
+        assert ed.synthesized_signal.shape == (k*2, n_chans, n, n)
+        assert ed.synthesized_eigenindex.allclose(torch.cat((torch.arange(k), torch.arange(n**2 - k, n**2))))
+        assert len(ed.synthesized_eigenvalues) == 2*k
 
-    def test_method_equivalence(self):
+    def test_method_svd(self):
+        n, k = 30, 10
+        n_chans = 1  # TODO color
+        ed = get_synthesis_object(im_dim=n)
+        ed.synthesize(k=k, method='svd')
+        assert ed.synthesized_signal.shape == (k, n_chans, n, n)
+        assert ed.synthesized_eigenindex.allclose(torch.arange(k))
+        assert len(ed.synthesized_eigenvalues) == k
 
+    def test_method_accuracy(self):
+        # test pow and svd against ground-truth jacobian (exact) method
         e_jac = get_synthesis_object(im_dim=SMALL_DIM)
         e_pow = get_synthesis_object(im_dim=SMALL_DIM)
+        e_svd = get_synthesis_object(im_dim=SMALL_DIM)
 
-        e_jac.synthesize(method='exact', store_progress=True)
-        e_pow.synthesize(method='power', n_steps=500, store_progress=True)
-
-        print(e_pow.synthesized_eigenvalues.shape)
-        print(e_pow.synthesized_eigenvalues[0], e_pow.synthesized_eigenvalues[1])
-        print(e_jac.synthesized_eigenvalues[0], e_jac.synthesized_eigenvalues[-1])
+        k_pow, k_svd = 10, 75
+        e_jac.synthesize(method='exact')
+        e_pow.synthesize(k=k_pow, method='power', max_steps=300)
+        e_svd.synthesize(k=k_svd, method='svd')
 
         assert e_pow.synthesized_eigenvalues[0].isclose(e_jac.synthesized_eigenvalues[0], atol=1e-3)
-        assert e_pow.synthesized_eigenvalues[1].isclose(e_jac.synthesized_eigenvalues[-1], atol=1e-2)
-
-        fig_max = e_pow.plot_loss(0)
-        fig_max.show()
-
-        fig_min = e_pow.plot_loss(-1)
-        fig_min.show()
+        assert e_pow.synthesized_eigenvalues[-1].isclose(e_jac.synthesized_eigenvalues[-1], atol=1e-2)
+        assert e_svd.synthesized_eigenvalues[0].isclose(e_jac.synthesized_eigenvalues[0], atol=1e-3)
 
     @pytest.mark.parametrize("color", [False, True])
     def test_display(self, color):
         e_pow = get_synthesis_object(im_dim=SMALL_DIM, color=color)
-        e_pow.synthesize(method='power', n_steps=50, store_progress=True)
+        e_pow.synthesize(method='power', max_steps=50, store_progress=True)
```

Comment:
I think we want some other tests here:
 - different methods
 - different number of eigenvectors
 - edge cases: `max_steps=0`, `store_progress=False` (ensure those fail in the expected ways)

Reply:
done.